File: solutions/csharp/scrabble-score/1/ScrabbleScore.cs
using System;

public static class ScrabbleScore
{
    public static int Score(string input)
    {
        int score = 0;
        input = input.ToUpper();
        foreach (char ch in input) {
            if ("A, E, I, O, U, L, N, R, S, T".Contains(ch)) score += 1;
            if ("D, G".Contains(ch)) score += 2;
            if ("B, C, M, P".Contains(ch)) score += 3;
            if ("F, H, V, W, Y".Contains(ch)) score += 4;
            if (ch=='K') score += 5;
            if ("J, X".Contains(ch)) score += 8;
            if ("Q, Z".Contains(ch)) score += 10;
        }
        return score;
    }
}

File: solutions/cpp/scrabble-score/1/scrabble_score.cpp
#include "scrabble_score.h"

namespace scrabble_score {
    int score(const char* word) {
        int s=0;
        for (size_t i=0; i<strlen(word); i++) {
            char ch = toupper(word[i]);
            if (strchr("A, E, I, O, U, L, N, R, S, T",ch)) s+=1;
            if (strchr("D, G",ch)) s+=2;
            if (strchr("B, C, M, P",ch)) s+=3;
            if (strchr("F, H, V, W, Y",ch)) s+=4;
            if (ch=='K') s+=5;
            if (strchr("J, X",ch)) s+=8;
            if (strchr("Q, Z",ch)) s+=10;
        }
        return s;
    }
}  // namespace scrabble_score
